Index: src/ca/slashdev/bb/tasks/RapcTask.java
===================================================================
--- src/ca/slashdev/bb/tasks/RapcTask.java	(revision 62)
+++ src/ca/slashdev/bb/tasks/RapcTask.java	(working copy)
@@ -568,6 +568,14 @@
             java.createArg().setFile(new File(file));
          }
       }
+
+      // Set the default file encoding to match its default value on Windows,
+      // so that non-Windows build environments will process resource files
+      // correctly.
+      Environment.Variable fileEncodingProperty = new Environment.Variable();
+      fileEncodingProperty.setKey("file.encoding");
+      fileEncodingProperty.setValue("windows-1252");
+      java.addSysproperty(fileEncodingProperty);
       
       log(java.getCommandLine().toString(), Project.MSG_DEBUG);
       java.execute();
Index: src/ca/slashdev/bb/tasks/SigtoolTask.java
===================================================================
--- src/ca/slashdev/bb/tasks/SigtoolTask.java	(revision 62)
+++ src/ca/slashdev/bb/tasks/SigtoolTask.java	(working copy)
@@ -25,6 +25,7 @@
 import org.apache.tools.ant.BuildException;
 import org.apache.tools.ant.Project;
 import org.apache.tools.ant.taskdefs.Java;
+import org.apache.tools.ant.types.Environment;
 import org.apache.tools.ant.types.Resource;
 import org.apache.tools.ant.types.ResourceCollection;
 import org.apache.tools.ant.types.resources.FileResource;
@@ -185,6 +186,12 @@
       for (String file : codFiles.list()) {
          java.createArg().setFile(new File(file));
       }
+
+      // Set the default file encoding to match its default value on Windows.
+      Environment.Variable fileEncodingProperty = new Environment.Variable();
+      fileEncodingProperty.setKey("file.encoding");
+      fileEncodingProperty.setValue("windows-1252");
+      java.addSysproperty(fileEncodingProperty);
       
       if (java.executeJava() == 0) {
          for (String file : codFiles.list()) {
